0013-RS-pubsub-subscription-streaming.md

```diff
@@ -0,0 +1,159 @@
+  // id is the unique identifier for the subscription request.
+  string id = 1 [json_name = "id"];
+
+  // status is the result of the subscription request.
+  TopicEventResponse status = 2 [json_name = "status"];
+}
+```
+
+When an application wishes to subscribe to a topic it will initiate a stream with `SubscribeTopicEventsRequest`, and `Send` the initial request `SubscribeTopicEventsSubscribeRequest` containing the options for the subscription.
+Daprd will then setup the machinery to add this gRPC RPC stream to the set of subscribers.
+The request contains no route or path matching configuration as all events will be sent on this stream.
+Subscription gRPC streams are the highest priority when Daprd determines which publisher a message should be sent to.
+Only a single PubSub Topic pair may be subscribed at a single time with this API.
+If the first message sent to the server is not the initial request, the RPC will return an error.
+If any subsequent messages are not `SubscribeTopicEventsResponse` messages, the RPC will return an error.
+
+When a message is published to the topic, Daprd will send a `TopicEventRequest` message on the stream containing the message payload and metadata.
+After the application has processed the message, it will send to the server a `SubscribeTopicEventsResponse` containing the `id` of the message and the `status` of the message processing.
+Since multiple messages can be sent and processed in the application at the same time, the event `id` is used by the server to track the status of each individual event.
+An event topic response will follow the timeout resiliency as currently exist for subscriptions.
+
+Client code:
+
+```go
+	stream, _ := client.SubscribeTopicEvents(ctx)
+	stream.Send(&rtv1.SubscribeTopicEventsRequest{
+		SubscribeTopicEventsRequestType: &rtv1.SubscribeTopicEventsRequest_Request{
+			Request: &rtv1.SubscribeTopicEventsSubscribeRequest{
+				PubsubName: "mypub", Topic: "a",
+			},
+		},
+	})
+
+	client.PublishEvent(ctx, &rtv1.PublishEventRequest{
+		PubsubName: "mypub", Topic: "a",
+		Data:            []byte(`{"status": "completed"}`),
+		DataContentType: "application/json",
+	})
+
+	event, _ := stream.Recv()
+	stream.Send(&rtv1.SubscribeTopicEventsRequest{
+		SubscribeTopicEventsRequestType: &rtv1.SubscribeTopicEventsRequest_EventResponse{
+			EventResponse: &rtv1.SubscribeTopicEventsResponse{
+				Id:     event.Id,
+				Status: &rtv1.TopicEventResponse{Status: rtv1.TopicEventResponse_SUCCESS},
+			},
+		},
+	})
+
+	stream.CloseSend()
+```
+
+### HTTP (WebSockets)
+
+Along with a gRPC based streaming API, a WebSocket based HTTP equivalent API will be implemented.
+Much like the gRPC API, the HTTP based WebSocket API will follow an initial request-response handshake, followed by a stream of messages to the client with status responses by the client, indexed by the message ID.
+The same proto types as using in the gRPC API (but in JSON blobs) will be used for the HTTP API.
```

Comment:
can you add some more details on the messages and their actual JSON structure here in the document?